Polyfill/CommandPrediction.cs

```diff
@@ -0,0 +1,135 @@
+#if LEGACY
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Management.Automation.Language;
+
+namespace System.Management.Automation.Subsystem
+{
+    /// <summary>
+    /// The class represents the prediction result from a predictor.
+    /// </summary>
+    public sealed class PredictionResult
+    {
+        /// <summary>
+        /// Gets the Id of the predictor.
+        /// </summary>
+        public Guid Id { get; }
+
+        /// <summary>
+        /// Gets the name of the predictor.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// Gets the suggestions.
+        /// </summary>
+        public IReadOnlyList<PredictiveSuggestion> Suggestions { get; }
+
+        internal PredictionResult(Guid id, string name, List<PredictiveSuggestion> suggestions)
+        {
+            Id = id;
+            Name = name;
+            Suggestions = suggestions;
+        }
+    }
+
+    /// <summary>
+    /// The class represents a predictive suggestion generated by a predictor.
+    /// </summary>
+    public sealed class PredictiveSuggestion
+    {
+        /// <summary>
+        /// Gets the suggestion.
+        /// </summary>
+        public string SuggestionText { get; }
+
+        /// <summary>
+        /// Gets the tooltip of the suggestion.
+        /// </summary>
+        public string ToolTip { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PredictiveSuggestion"/> class.
+        /// </summary>
+        /// <param name="suggestion">The predictive suggestion text.</param>
+        public PredictiveSuggestion(string suggestion)
```

Comment:
Consider decorating statically visible members with `HiddenAttribute`.  Since these will be visible from PowerShell, adding `Hidden` will reduce the chance of folks trying to use it.

Reply:
`HiddenAttribute` added to properties and methods of the stub types.